python/orion_finance_sdk_py/contracts.py

```
@@ -1657,83 +1680,155 @@ def get_portfolio_pct_tvl(self) -> dict[str, float]:
-        ).build_transaction({"from": account.address, "nonce": nonce})
+        setter = getattr(self.contract.functions, setter_name)
+        tx = setter(Web3.to_checksum_address(access_control_address)).build_transaction(
+            {"from": account.address, "nonce": nonce}
+        )
 
         signed = account.sign_transaction(tx)
         tx_hash = self.w3.eth.send_raw_transaction(signed.raw_transaction)
         tx_hash_hex = tx_hash.hex()
         receipt = self._wait_for_transaction_receipt(tx_hash_hex)
+        if receipt["status"] != 1:
+            raise Exception(f"Transaction failed with status: {receipt['status']}")
```

Ruff (0.16.3)

[warning] Create your own exception

(TRY002)